cdap-data-fabric/src/main/java/co/cask/cdap/data/runtime/AbstractTransactionSystemClientProvider.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.data.runtime;
+
+import co.cask.cdap.common.conf.CConfiguration;
+import co.cask.cdap.common.conf.Constants;
+import com.google.inject.Provider;
+import org.apache.tephra.TransactionSystemClient;
+
+/**
+ * Class to provide the {@link TransactionSystemClient}.
+ */
+abstract class AbstractTransactionSystemClientProvider implements Provider<TransactionSystemClient> {
+  private final CConfiguration cConf;
+
+  AbstractTransactionSystemClientProvider(CConfiguration cConf) {
+    this.cConf = cConf;
+  }
+
+  @Override
+  public TransactionSystemClient get() {
+    String storageImpl = cConf.get(Constants.Dataset.DATA_STORAGE_IMPLEMENTATION);
+    if (storageImpl == null) {
+      throw new IllegalStateException("No storage implementation is specified in the configuration file");
```

Comment:
We should add more info about action to be taken if this happens: `Please specify storage implementatio nto use in data.storage.implementation property of cdap-site.xml.` or something similar